src/lib/testing/cypress/mockutility.ts

```diff
@@ -0,0 +1,318 @@
+                  const routeDef: IRouteDef = {
+                    method,
+                    url,
+                  };
+
+                  cy.now('log', 'Registering alias route:');
+                  cy.now('log', 'url/method: ' + url + SLASH + method);
+                  cy.now('log', 'alias: ' + fixtureRouteDef.alias);
+                  cy.route(routeDef).as(fixtureRouteDef.alias);
+                }
+              });
+            });
+          });
+        });
+      },
+    );
+  }
+
+  registerMockRoute(url: string, method: string, fixtureRouteDef: IFixtureRouteDef): void {
+    const routeDef: IRouteDef = {
+      url,
+      method,
+      status: fixtureRouteDef.status,
+      headers: fixtureRouteDef.headers,
+      response: fixtureRouteDef.response,
+      onResponse: (xhr: any) => {
+        const responseURL: URL = new URL(xhr.url);
+        const responseKey: string = responseURL.pathname + PIPE + xhr.method;
+        const fixtureRouteDefs: IFixtureRouteDef[] = this.urlMethodFixtureMap.get(responseKey);
+        // If more routes exist, shift old fixture, and re-assign
+        if (fixtureRouteDefs.length > 1) {
+          fixtureRouteDefs.shift();
+          const updatedFixtureRouteDef: IFixtureRouteDef = fixtureRouteDefs[0];
+          this.registerMockRoute(url, method, updatedFixtureRouteDef);
+        }
+      },
+    };
+
+    cy.now('log', 'Registering mock route:');
+    cy.now('log', 'url/method: ' + url + SLASH + method);
+    cy.now('log', 'alias: ' + fixtureRouteDef.alias);
+    cy.now('log', 'status: ' + fixtureRouteDef.status);
+    cy.now('log', 'headers: ' + JSON.stringify(fixtureRouteDef.headers));
+    cy.now('log', 'response: ' + JSON.stringify(fixtureRouteDef.response));
+    cy.route(routeDef).as(fixtureRouteDef.alias);
+  }
+
+  recordResponse(xhr: any): void {
+    // Manipulate URL value to strip host info
+    const url: string = xhr.url.substring(Cypress.config('baseUrl').length);
+    const method: string = xhr.method;
+
+    // Remove black listed headers
+    const headers: any = {};
+    Object.keys(xhr.response.headers).forEach((key: string) => {
+      if (allowedHeaders.indexOf(key) !== -1) {
+        headers[key] = xhr.response.headers[key];
+      }
+    });
+
+    const fixtureRouteDef: IFixtureRouteDef = {
+      alias: url + PIPE + method,
+      status: xhr.status,
+      headers,
+      response: xhr.response.body,
+    };
+
+    if (!this.recordedURLMethodFixtureMap[url]) {
+      this.recordedURLMethodFixtureMap[url] = {};
+    }
+    const byMethodMap: Map<string, IFixtureRouteDef[]> = this.recordedURLMethodFixtureMap[url];
+    if (!byMethodMap[method]) {
+      byMethodMap[method] = [];
+    }
+    cy.now('log', 'Recording response:');
+    cy.now('log', 'url/method: ' + url + SLASH + method);
+    cy.now('log', 'response: ' + JSON.stringify(fixtureRouteDef));
+    byMethodMap[method].push(fixtureRouteDef);
+  }
+
+  waitForAngular(): Cypress.Chainable {
```

Comment:
Already in the other utilities file, can be removed from this file

Reply:
Done